src/DotNetWorker.Core/FunctionMetadata/DefaultFunctionMetadata.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Azure.Functions.Worker.Core.FunctionMetadata
+{
+    /// <summary>
+    /// Local representation of FunctionMetadata
+    /// </summary>
+    public class DefaultFunctionMetadata : IFunctionMetadata
+    {
+        /// <summary>
+        /// Creates a DefaultFunctionMetadata with the minimum required properties for successful function loading.
+        /// </summary>
+        /// <param name="functionId">Unique ID associated with each function.</param>
+        /// <param name="language">The language that the function is written in.</param>
+        /// <param name="name">The function name</param>
+        /// <param name="entryPoint">The function entrypoint</param>
+        /// <param name="rawBindings">List of the functions bindings in JSON string format</param>
+        /// <param name="scriptFile">The path to the function app assembly</param>
+        public DefaultFunctionMetadata(string functionId, string language, string name, string entryPoint, IList<string> rawBindings, string scriptFile)
+        {
+            FunctionId = functionId;
+            Language = language;
+            Name = name;
+            EntryPoint = entryPoint;
+            RawBindings = rawBindings;
+            ScriptFile = scriptFile;
+        }
+
+        /// <summary>
+        /// Unique ID associated with each function.
+        /// </summary>
+        public string FunctionId { get; set; }
```

Comment:
Curious if there is a way to avoid the redundant declaration of properties between `DefaultFunctionMetadata` and `IFunctionMetadata`. One concern with redundancy is - if I change a property name in interface, will I need to remember to update it in the class or will I get the compilation error (and vice-versa scenario).

May be recommendation provided in this doc can help - https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/interface-properties.